packages/flutter_tools/test/general.shard/asset_bundle_variant_test.dart

```diff
@@ -9,84 +9,146 @@ import 'package:file/memory.dart';
 
 import 'package:flutter_tools/src/asset.dart';
 import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/base/platform.dart';
+import 'package:flutter_tools/src/base/user_messages.dart';
+import 'package:flutter_tools/src/cache.dart';
 
-import 'package:flutter_tools/src/globals.dart' as globals;
+import 'package:flutter_tools/src/project.dart';
 
 import '../src/common.dart';
-import '../src/context.dart';
 
 void main() {
-  String fixPath(String path) {
-    // The in-memory file system is strict about slashes on Windows being the
-    // correct way so until https://github.com/google/file.dart/issues/112 is
-    // fixed we fix them here.
-    // TODO(dantup): Remove this function once the above issue is fixed and
-    // rolls into Flutter.
-    return path.replaceAll('/', globals.fs.path.separator);
+
+  Future<Map<String, List<String>>> extractAssetManifestFromBundle(ManifestAssetBundle bundle) async {
+    final String manifestJson = utf8.decode(await bundle.entries['AssetManifest.json']!.contentsAsBytes());
+    final Map<String, dynamic> parsedJson = json.decode(manifestJson) as Map<String, dynamic>;
+    final Iterable<String> keys = parsedJson.keys;
+    final Map<String, List<String>> parsedManifest = <String, List<String>> {
+      for (final String key in keys) key: List<String>.from(parsedJson[key] as List<dynamic>),
+    };
+    return parsedManifest;
   }
 
   group('AssetBundle asset variants', () {
-    late FileSystem testFileSystem;
-    setUp(() async {
-      testFileSystem = MemoryFileSystem(
-        style: globals.platform.isWindows
-          ? FileSystemStyle.windows
-          : FileSystemStyle.posix,
-      );
-      testFileSystem.currentDirectory = testFileSystem.systemTempDirectory.createTempSync('flutter_asset_bundle_variant_test.');
-    });
+    testWithoutContext('Only images in folders named with device pixel ratios (e.g. 2x, 3.0x) should be considered as variants of other images', () async {
+      final MemoryFileSystem fs = MemoryFileSystem.test();
```

Comment:
Can you set the `MemoryFileSystem.style` to Windows if we're running on a Windows bot?

Reply:
Hmm, we could; but that would make the tests a bit uglier. To my knowledge, the asset manifest we generates always uses POSIX-style paths. That means that each test case will have to create both representations (one for setup using `join` and one for the assertion).

Take this assertion for example:

https://github.com/andrewkolos/flutter/blob/f7dccd2bd2414769bd75fa8bf178cf10ecc5010d/packages/flutter_tools/test/general.shard/asset_bundle_variant_test.dart#L130


Instead of being able to reuse `secondLevelImage` (the name pointing to the asset's path) directly, I would have to do something else, such as hard-code a POSIX version of the path.

Reply:
Never mind, I failed to consider that I could just fix the paths when making the file.create calls and the rest of the code could stay the same.